plugins/currency/nanopool/nanopool_

```diff
@@ -0,0 +1,290 @@
+        hashrate: link to nanopool_hashrate         Show current calculated hashrate
+        avghashrate: link to nanopool_avghashrate   Show average hashrate of last hour
+        worker: link to nanopool_worker             Show worker data
+
+
+Thanks to the authors of other Python munin plugins. I've used some of
+them as inspiring example.
+
+=head1 VERSION
+1.0.0
+
+=head1 AUTHOR
+L<Ralf Geschke|https://github.com/geschke>
+
+=head1 LICENSE
+
+This program is free software: you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+=head1 MAGIC MARKERS
+
+#%# family=contrib
+#%# capabilities=suggest
+
+=cut
+"""
+
+from __future__ import print_function
+
+import os
+import sys
+import json
+
+try:
+    # Python 3
+    from urllib.request import Request
+    from urllib.request import urlopen
+except:
+    # Python 2
+    from urllib2 import Request
+    from urllib2 import urlopen
+   
+
+
+def define_graph_types():
+    graph_types = {
+        "balance" : [
+            {
+                "title" : "Balance of {0}".format(account_address),
+                "type" : "GAUGE",
+                "args" : "--base 1000 -l 0",
+                "fields" : ["ETH"],
+                "scale": "no",
+                "info": "Balance in ETH"
+            }
+        ],
+        "hashrate" : [
+            {
+                "title" : "Hashrate of {0}".format(account_address),
+                "type" : "GAUGE",
+                "args" : "--base 1000 -l 0",
+                "fields" : ["hashrate"],
+                "info": "Current Calculated Hashrate in Mh/s"
+            }
+        ],
+        "avghashrate" : [
+            {
+                "title" : "Average Hashrate of {0}".format(account_address),
+                "type" : "GAUGE",
+                "args" : "--base 1000 -l 0",
+                "fields" : ["avg_hashrate"],
+                "info": "Average Hashrate of last hour in Mh/s"
+
+            }
+        ],
+        "worker" : [
+            {
+                "title" : "Worker"
+            }
+        ]
+    }
+    return graph_types
+
+
+def request_data():
+
+    url = "https://api.nanopool.org/v1/eth/user/{0}".format(account_address)
+
+    req = Request(url)
+    # this fake is necessary to get values, otherwise the request ends in a 403 error
+    req.add_header('User-Agent', 'Nozilla/5.0')
+    try:
+        txt = urlopen(req).read()
+    except Exception as err:
```

Comment:
Please avoid generic exception traps.

Here you could either use `URLError` and `HTTPError` (with their respective imports for python2 and python3) or you could simpliy pick `OSError` (which is the parent of both of these exceptions).